dev/tests/api-functional/testsuite/Magento/GraphQl/GiftMessage/Order/GiftMessageTest.php

```diff
@@ -0,0 +1,108 @@
+<?php
+/**
+ * Copyright © Magento, Inc. All rights reserved.
+ * See COPYING.txt for license details.
+ */
+declare(strict_types=1);
+
+namespace Magento\GraphQl\GiftMessage\Order;
+
+use Exception;
+use Magento\Framework\Exception\AuthenticationException;
+use Magento\Integration\Api\CustomerTokenServiceInterface;
+use Magento\TestFramework\Helper\Bootstrap;
+use Magento\TestFramework\TestCase\GraphQlAbstract;
+
+class GiftMessageTest extends GraphQlAbstract
+{
+    /**
+     * @var CustomerTokenServiceInterface
+     */
+    private $customerTokenService;
+
+    protected function setUp()
+    {
+        parent::setUp();
+        $this->customerTokenService = Bootstrap::getObjectManager()->get(CustomerTokenServiceInterface::class);
+    }
+
+    /**
+     * @magentoConfigFixture default_store sales/gift_options/allow_order 1
+     * @magentoConfigFixture default_store sales/gift_options/allow_items 1
+     * @magentoApiDataFixture Magento/Customer/_files/customer.php
+     * @magentoApiDataFixture Magento/GiftMessage/_files/customer/order_with_message.php
+     * @throws AuthenticationException
+     * @throws Exception
+     */
+    public function testGiftMessageForOrder()
+    {
+        $query =
```

Comment:
To add some maintainability to the test, you may extract this query into a separate private method since it's the same for both test cases

Reply:
**Done**